tests/test_execution.py

```diff
@@ -245,3 +245,124 @@ def test_composite_traintuples_execution(factory, session):
     assert set([composite_traintuple_1.key, composite_traintuple_2.key]).issubset(
         composite_traintuple_keys
     )
+
+
+def test_aggregate_traintuples(factory, session):
+    """Aggregate traintuples."""
+    number_of_traintuples_to_aggregate = 3
+    spec = factory.create_dataset()
+    dataset = session.add_dataset(spec)
+
+    train_data_samples = []
+    for i in range(number_of_traintuples_to_aggregate):
+        spec = factory.create_data_sample(test_only=False, datasets=[dataset])
+        data_sample = session.add_data_sample(spec)
+        train_data_samples.append(data_sample)
+
+    spec = factory.create_objective(
+        dataset=dataset,
+        data_samples=[],
+    )
+    objective = session.add_objective(spec)
+
+    spec = factory.create_algo()
+    algo = session.add_algo(spec)
+
+    # add traintuples
+    traintuples = []
+    for data_sample in train_data_samples:
+        spec = factory.create_traintuple(
+            algo=algo,
+            objective=objective,
+            dataset=dataset,
+            data_samples=[data_sample],
+        )
+        traintuple = session.add_traintuple(spec).future().wait()
+        traintuples.add(traintuple)
+
+    spec = factory.create_aggregate_algo()
+    aggregate_algo = session.add_aggregate_algo(spec)
+
+    spec = factory.create_aggregatetuple(
+        algo=aggregate_algo,
+        objective=objective,
+        worker=session.node_id,
+        in_models_keys=traintuples,
+    )
+    aggregatetuple = session.add_aggregatetuple(spec).future().wait()
+    assert aggregatetuple.status == 'done'
+
+
+def test_aggregate_composite_traintuples(factory, session_1, session_2):
+    """Do 2 rounds of composite traintuples aggregations on multiple nodes.
+
+    This test refers to the model composition use case.
```

Comment:
Would there be a way to include a quick description of the use case? Just to make clear what we're executing here. Maybe some asciiart? I'm half joking about this, since we're creating tuple-graphs, it'd be very nice to have a way to represent these graphs inside the code.

Reply:
I fully agree, this is quite hard to understand what is the goal of this test.

I added a detailed description of the compute plan.